.github/workflows/macos.yml

```diff
@@ -71,9 +84,36 @@ jobs:
 
       - run: make
 
+      # Launchable requires Python and Java
+      # https://www.launchableinc.com/docs/resources/cli-reference/
+      - name: Set up Python
+        uses: actions/setup-python@871daa956ca9ea99f3c3e30acb424b7960676734 # v5.0.0
+        with:
+          python-version: "3.10"
+      - name: Set up JDK 17
+        uses: actions/setup-java@7a445ee88d4e23b52c33fdc7601e40278616c7f8 # v4.0.0
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+      - name: Setup Launchable
+        id: setup_launchable
+        run: |
```

Comment:
I would suggest inserting the `set -ex` here on the `run |` below.

Because seeing the log in this part, there are errors (exceptions) maybe cby the launchable record build ...`. But maybe GitHub Actions failed to catch the error exit status, and moved to the next command. And I also see the `set -x` is useful to print the output after each command.

Reply:
Updated

Reply:
Thanks! I think the output became much more readable and maintainable with the `set -ex`.

You see the exception messages from Java from line 87 to line 2703 below. Do you know why the exceptions happened? Is it an expected behavior? Interestingly it seems that the `launchable record build ...` still returns the exit status zero even when printing the messages about the exceptions.

https://github.com/ruby/ruby/actions/runs/7912934050/job/21599616384?pr=9777#step:10:85

```
+ launchable record build --name c9b5358d2572b455dd2f378d00d307c4d067a3b7
Feb 15, 2024 8:22:30 AM com.launchableinc.ingest.commits.CommitGraphCollector$ByRepository transform
WARNING: Failed to process a change to a file
 java.io.UncheckedIOException: org.eclipse.jgit.errors.MissingObjectException: Missing blob c7f57385b32005caa5512d52f72135a5dc1df4de
	at com.launchableinc.ingest.commits.CountingDiffFormatter.process(CountingDiffFormatter.java:38)
	at com.launchableinc.ingest.commits.CommitGraphCollector$ByRepository.transform(CommitGraphCollector.java:410)
	at com.launchableinc.ingest.commits.CommitGraphCollector$ByRepository.transfer(CommitGraphCollector.java:343)
	at com.launchableinc.ingest.commits.CommitGraphCollector.transfer(CommitGraphCollector.java:195)
	at com.launchableinc.ingest.commits.CommitGraphCollector.transfer(CommitGraphCollector.java:129)
	at com.launchableinc.ingest.commits.CommitIngester.run(CommitIngester.java:149)
	at com.launchableinc.ingest.commits.CommitIngester.main(CommitIngester.java:77)
Caused by: org.eclipse.jgit.errors.MissingObjectException: Missing blob c7f57385b32005caa5512d52f72135a5dc1df4de
	at org.eclipse.jgit.internal.storage.file.WindowCursor.open(WindowCursor.java:137)
	at org.eclipse.jgit.diff.ContentSource$ObjectReaderSource.open(ContentSource.java:112)
	at org.eclipse.jgit.diff.ContentSource$Pair.open(ContentSource.java:257)
	at org.eclipse.jgit.diff.DiffFormatter.open(DiffFormatter.java:1074)
	at org.eclipse.jgit.diff.DiffFormatter.createFormatResult(DiffFormatter.java:1001)
	at org.eclipse.jgit.diff.DiffFormatter.format(DiffFormatter.java:704)
	at com.launchableinc.ingest.commits.CountingDiffFormatter.process(CountingDiffFormatter.java:36)
	... 6 more
...
```

Reply:
Thank you for bringing this to my attention. While I'm not currently familiar with these specific warnings, it seems that they haven't obstructed the successful execution of the `launchable record build` command, which is reassuring. I'll be sure to reach out to the Launchable CLI team to investigate the reasons behind these warnings. Rest assured, I'll keep you updated once I have more information.

Reply:
All right. Thank you for your action to investigate the warnings!

Reply:
Updated

Reply:
Thanks! I confirmed the link was added in the latest commit on this PR.